src/lang/usages/mod.rs

```diff
@@ -192,9 +192,35 @@ impl<'db> FindUsages<'db> {
         let found_symbol_declaration =
             SymbolSearch::find_declaration(self.db, &identifier).map(|ss| ss.def);
 
+        // We check if a SymbolDef is in a MemberDef in order to see if we're dealing with a member
+        // of a struct annotated with a macro. Non-annotated structs do not need this.
+        fn as_member<'a>(sym: &'a Option<SymbolDef<'a>>) -> Option<&'a MemberDef<'a>> {
+            match sym {
+                Some(SymbolDef::Member(m)) => Some(m),
+                _ => None,
+            }
+        }
+
+        // We check if a symbol is a Member of a struct, then compare its item_id with a context
+        // of found symbol definitions and declarations
+        let symbol_in_context = match &self.symbol {
+            SymbolDef::Member(member_def) => {
+                let id = member_def.struct_item().lookup_item_id();
+
+                let in_context = |m: &MemberDef| m.struct_item().context_items().contains(&id);
```

Comment:
This is a bit of a weird heuristic - why would one compare `lookup_item_id` to check if the definition/declaration is equal?